pkg/neonvm/controllers/vm_controller.go

```diff
@@ -864,12 +865,45 @@ func runnerStatus(pod *corev1.Pod) runnerStatusKind {
 	case corev1.PodFailed:
 		return runnerFailed
 	case corev1.PodRunning:
-		return runnerRunning
+		return isRunnerPodReady(pod)
 	default:
 		panic(fmt.Errorf("unknown pod phase: %q", pod.Status.Phase))
 	}
 }
 
+// isRunnerPodReady returns whether the runner pod is ready respecting the readiness probe of its containers.
+func isRunnerPodReady(pod *corev1.Pod) runnerStatusKind {
+	// if the pod has no container statuses, we consider it pending
+	if len(pod.Status.ContainerStatuses) == 0 {
+		return runnerPending
+	}
+
+	// first we check if the pod is a target pod for the migration
+	// in that case we don't wait for the readiness probe
+	// from neonvm-daemon because qemu started
+	// in incoming migration mode
+	for _, containerSpec := range pod.Spec.Containers {
+		if containerSpec.Name == "neonvm-runner" {
+			for _, env := range containerSpec.Env {
+				if env.Name == "RECEIVE_MIGRATION" && env.Value == "true" {
+					return runnerRunning
```

Comment:
suggestion(non-blocking): Maybe extract the "is this a target pod" check into a separate function?

Reply:
done